### DISCUSSION TITLE
How can I have tabnine autocompletion?

### DISCUSSION THREAD

**Discussion by ghost** [NvChad/NvChad#853] [Q&A]
Where do I have to copy this? Will it conflict with the current completion?

The tabnine installation for [nvim-cmp using packer](https://github.com/tzachar/cmp-tabnine#install) is:
```
return require("packer").startup(
 function(use)
 	use "hrsh7th/nvim-cmp" --completion
 	use {'tzachar/cmp-tabnine', run='./install.sh', requires = 'hrsh7th/nvim-cmp'}
 end
)
```
And later, enable the plugin with:
```
require'cmp'.setup {
 sources = {
 	{ name = 'cmp_tabnine' },
 },
}
```
Setup:
```
local tabnine = require('cmp_tabnine.config')
tabnine:setup({
	max_lines = 1000;
	max_num_results = 20;
	sort = true;
	run_on_every_keystroke = true;
	snippet_placeholder = '..';
	ignored_file_types = { -- default is not to ignore
		-- uncomment to ignore in lua:
		-- lua = true
	};
})
```

**Comment by siduck** [comment]
first add tabnine https://nvchad.github.io/config/Custom%20config#add-plugins 

then override cmp config 
https://nvchad.github.io/config/Custom%20config#override-default-config-of-a-plugin

**Reply by ghost** [reply]
There must be something I'm doing wrong. When I do `:PackerSync` nothing happens.

    -- NvChad/lua/custom/plugins/init.lua
    return {
       ["tzachar/cmp-tabnine"] = {
          after = "cmp-path",
          config = function()
             require "custom.plugins.tabnine"
          end,
          run='./install.sh',
       },
    }

    -- NvChad/lua/custom/plugins/tabnine.lua
    require('cmp_tabnine.config').setup({
       max_lines = 1000;
       max_num_results = 20;
       sort = true;
       run_on_every_keystroke = true;
       snippet_placeholder = '..';
       ignored_file_types = { -- default is not to ignore
               -- uncomment to ignore in lua:
               -- lua = true
       };
       show_prediction_strength = false;
    })

    -- NvChad/lua/custom/chardrc.lua
    M.plugins = {
       override = {
          ["hrsh7th/nvim-cmp"] = {
             sources = {
                { name = "nvim_lsp" },
                { name = "luasnip" },
                { name = "buffer" },
                { name = "nvim_lua" },
                { name = "path" },
                { name = "cmp-tabnine" },
              },
          }
       }
    }

**Reply by nxlzto** [reply]
{ name = "cmp-tabnine" } to { name = "cmp_tabnine" },
